py/dml/c_backend.py

```diff
@@ -917,120 +925,145 @@ def generate_implements(code, device):
         except DMLError as e:
             report(e)
 
-def simple_event_fun_deconstruct_data(method, data):
-    if method.dimensions > 0 and len(method.inp) > 0:
-        out(f'ASSERT(SIM_attr_list_size({data}) == 2);\n')
-        out(f'attr_value_t indices = SIM_attr_list_item({data}, 0);\n')
-        out(f'attr_value_t params = SIM_attr_list_item({data}, 1);\n')
-    elif method.dimensions > 0 or len(method.inp) > 0:
-        varname = 'indices' if method.dimensions > 0 else 'params'
-        out(f'attr_value_t {varname} = {data};\n')
-
-    if method.dimensions > 0:
-        out(f'ASSERT(SIM_attr_list_size(indices) == {method.dimensions});\n')
+def generate_simple_event_only_domains_funs():
+    start_function_definition(
+        'attr_value_t _simple_event_only_domains_get_value('
+        + 'conf_object_t *_obj, lang_void *data)')
+    out('{\n', postindent=1)
+    out('return _serialize_simple_event_data(0, NULL, _id_infos, '
+        + '(_simple_event_data_t *)data);\n')
+    out('}\n', preindent=-1)
+    splitting_point()
 
-    if len(method.inp) > 0:
-        out(f'ASSERT(SIM_attr_list_size(params) == {len(method.inp)});\n')
+    start_function_definition(
+        'lang_void *_simple_event_only_domains_set_value('
+        + 'conf_object_t *_obj, attr_value_t val)')
+    out('{\n', postindent=1)
+    out('_simple_event_data_t *out;\n')
+    out('set_error_t error = _deserialize_simple_event_data(NULL, 0, 0, NULL, '
+        + '&_id_info_ht, val, &out);\n')
+    out('if (error != Sim_Set_Ok) {\n', postindent=1)
+    out('out = NULL;\n')
+    out('}\n', preindent=-1)
+    out('return out;\n')
+    out('}\n', preindent=-1)
+    splitting_point()
 
 def generate_simple_events(device):
     handled = set()
     done = False
 
+    # TODO PURGE
```

Comment:
Refers to the quadratic nonsense with `handled` and `done`. Forgot doing this before I published the PR.